text/0052-resource-importing-support.md

```diff
@@ -0,0 +1,272 @@
+                                 that can't be identified will be skipped from
+                                 import. (default: false if stdin is a TTY, true
+                                 otherwise)
+```
+
+Sometimes, it is useful to be able to import (enroll/adopt/...) AWS resources,
+that were created by different means (manually in AWS console, using different
+orchestrator, etc), into a CDK stack. Some resources can simply be deleted and
+recreated by CDK, but for others, this is not convenient: Typically stateful
+resources like S3 Buckets, DynamoDB tables etc, cannot be easily deleted without
+an impact on the service.
+
+To import an existing resource into a CDK stack:
+
+- Run a `cdk diff` to ensure there are no pending changes to the CDK stack you
+  want to import resources into - if there are, apply/discard them first.
+- Add corresponding constructs for the resources to be added in your stack - for
+  example, for an S3 bucket, add something like
+  `new s3.Bucket(this, 'ImportedS3Bucket', {});` - **no other changes must be
+  done to the stack before the import is completed**.
+- Run `cdk import` command - if there are multiple stacks in the CDK app, pass a
+  specific stack name as an argument.
+- If the resource definition contains all information needed for the import,
+  this happens automatically (e.g. an `s3.Bucket` construct has an explicit
+  `bucketName` set), otherwise, CDK will display an interactive prompt to
+  provide neccessary identification information (e.g. the bucket name).
+- After `cdk import` reports success, the resource is managed by CDK. Any
+  subsequent changes to the construct configuration will be reflected on the
+  resource.
+
+---
+
+Ticking the box below indicates that the public API of this RFC has been
+signed-off by the API bar raiser (the `api-approved` label was applied to the
+RFC pull request):
+
+```
+[ ] Signed-off by API Bar Raiser @rix0rrr
+```
+
+## Public FAQ
+
+### What are we launching today?
+
+New CDK CLI sub-command `cdk import` that works just like `cdk deploy` but for
+newly added constructs in the stack, instead of creating new AWS resources, it
+attempts to import corresponding existing resources.
+
+### Why should I use this feature?
+
+When an already existing AWS resource needs to be brought under CDK management
+the most convenient way of achieving this is using this feature (`cdk import`).
+
+## Internal FAQ
+
+### Why are we doing this?
+
+This solution is exposing an existing [CloudFormation resource
+import](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/resource-import.html)
+feature in a CDK native way. CDK's goal is to provide convenient abstraction
+over sometimes complex CloudFormation aspects and this new feature is aligned
+with this goal. Additionally, other IaC orchestrators' also provide a
+corresponding feature, for example:
+
+  * [terraform import](https://www.terraform.io/cli/import)
```

Comment:
Oh excellent research! That clinches the command's name then :)

Reply:
cheers!